worker/dbaccessor/worker.go

```diff
@@ -266,6 +266,41 @@ func (w *dbWorker) Wait() error {
 	return w.catacomb.Wait()
 }
 
+// Report provides information for the engine report.
+func (w *dbWorker) Report() map[string]any {
+	w.mu.RLock()
+	defer w.mu.RUnlock()
+
+	// We need to guard against attempting to report when setting up or dying,
+	// so we don't end up panic'ing with missing information.
+	result := w.dbRunner.Report()
+
+	if w.dbApp == nil {
+		result["leader"] = ""
+		result["leader-role"] = ""
+		return result
+	}
+
+	ctx, cancel := w.scopedContext()
+	defer cancel()
+
+	var (
+		leader     string
+		leaderRole string
+	)
+	if client, err := w.dbApp.Client(ctx); err == nil {
+		if nodeInfo, err := client.Leader(ctx); err == nil {
+			leader = nodeInfo.Address
+			leaderRole = nodeInfo.Role.String()
+		}
+	}
+
+	result["leader"] = leader
```

Comment:
Is the nodeInfo.ID useful? Address is a bit of a pain, as you have to map it back.

Reply:
Node id isn't something that is exposed anywhere currently. Only the address is valuable here.

Reply:
I'll add it anyway.